### ISSUE TITLE
Add keyword argument to define the number of rows to show independent of the display size.

### ISSUE THREAD

**Issue by nathanrboyer** [ronisbr/PrettyTables.jl#101]
I love the new `vcrop_mode = :middle` feature, but I would like a way to define how many rows appear on either side of the ellipsis. `display_size` can achieve this indirectly, but the issue is that it requires you to define both dimensions. I would like to define row cropping while leaving column cropping as automatic. (`display_size = (12, _)` didn't work.)

Additionally, it would be much nicer if I could specify the number of rows to show in terms of row numbering instead of display size e.g. show two rows on either side of the ellipsis plus the header and title. The keyword argument could be implemented as an integer to keep it simple across all row cropping behaviors. The integer number of rows would be applied to both the top and bottom for `vcrop_mode = :middle`. Some people might want a tuple for asymmetric splitting, but I probably wouldn't use it that way.

An analogous keyword argument could be created to display discrete columns too, but I think it would be more commonly used in terms of rows. I am requesting this feature so that my table can be presented as part of a larger report and not use all of the vertical space itself.
 
![image](https://user-images.githubusercontent.com/65452054/100908141-29271580-3499-11eb-82e3-7d46cd19b3a6.png)

**Comment by ronisbr** [comment]
> I love the new `vcrop_mode = :middle` feature, but I would like a way to define how many rows appear on either side of the ellipsis. `display_size` can achieve this indirectly, but the issue is that it requires you to define both dimensions. I would like to define row cropping while leaving column cropping as automatic. (`display_size = (12, _)` didn't work.)
> 
> Additionally, it would be much nicer if I could specify the number of rows to show in terms of row numbering instead of display size e.g. show two rows on either side of the ellipsis plus the header and title. The keyword argument could be implemented as an `Integer` to keep it simple across all `vcrop_mode`s. The number of rows would be applied to both the top and bottom for `vcrop_mode = :middle`. Some people might want a tuple for asymmetric splitting in this case, but I probably wouldn't use it that way.
> 
> An analogous keyword argument could be created to display discrete columns too, but I think it would be more commonly used in terms of rows. I am requesting this feature so that my table can be presented as part of a larger report and not use all of the vertical space itself.
> 
> ![image](https://user-images.githubusercontent.com/65452054/100908141-29271580-3499-11eb-82e3-7d46cd19b3a6.png)

Nice! I am glad it is begin useful :)



> but I would like a way to define how many rows appear on either side of the ellipsis. `display_size` can achieve this indirectly, but the issue is that it requires you to define both dimensions. I would like to define row cropping while leaving column cropping as automatic. (`display_size = (12, _)` didn't work.)

If a dimension is negative, then it is ignored. Then, I think you can do what you want by `display_size = (12, -1)`.



> Additionally, it would be much nicer if I could specify the number of rows to show in terms of row numbering instead of display size e.g. show two rows on either side of the ellipsis plus the header and title. The keyword argument could be implemented as an `Integer` to keep it simple across all `vcrop_mode`'s. The number of rows would be applied to both the top and bottom for `vcrop_mode = :middle`. Some people might want a tuple for asymmetric splitting in this case, but I probably wouldn't use it that way.

There is a problem to define the number of rows to be shown before and after the ellipsis because we have no idea of the size of each column. It can be 4 or 400. If you want to reduce the number of columns / rows that will be shown, then I think it is better to use a filter (see `filters_row` and `filters_col` options).

**Comment by nathanrboyer** [comment]
>Then, I think you can do what you want by display_size = (12, -1). 

`display_size = (12, -1)` says "do not crop the columns" not "fit the columns to the display size" like I am asking for.

>There is a problem to define the number of rows to be shown before and after the ellipsis because we have no idea of the size of each row. It can be 1 line or 400 lines.

Can't multi-line rows be treated the same way as a single-line row? You can still print however many rows the user requests (multi- or single-line), and if it is too big then there will just be scroll bars.

![image](https://user-images.githubusercontent.com/65452054/100913835-4ad7cb00-34a0-11eb-99c1-6a794198d178.png)

**Comment by ronisbr** [comment]
> `display_size = (12, -1)` says "do not crop the columns" not "fit the columns to the display size" like I am asking for.

Oh, sorry, now I understand what do you want. Just pass `crop = :horizontal`. It will crop horizontally but not vertically. Is this what you want?

**Comment by nathanrboyer** [comment]
No, I would like a manual vertical crop and an automatic horizontal crop.

**Comment by ronisbr** [comment]
The units in `display_size` are the length and width of displays (the number of rows and the number of characters that can be fitter in a row). But I will create something so that `display_size` can use the current length or width of the terminal. As a workaround, you can do: `julia> pretty_table(A, display_size = (20, displaysize()[2]))`. Hence it will consider that the display size is 20 rows.

**Comment by nathanrboyer** [comment]
The workaround is not working properly for me in Visual Studio Code. The `displaysize()` function is not returning the correct values. I manually count 23 rows and visually estimate about 200 columns. There is no need to rush for a quick fix though. I can live without this for now.

![image](https://user-images.githubusercontent.com/65452054/100925107-8af27a00-34af-11eb-92a3-fd94a316fd36.png)

Thank you for explaining the units for `display_size`. It makes sense now. However, it still seems strange that the user cannot directly ask for some number of table rows to be displayed. I will have to take the number of table rows I want to display, add 1 for the ellipsis line, 3 for the borders, 2 for the header, 1 for the title, 1 for the footer, 1 for the newline, and 1 for the next "julia" prompt in order to get the correct number to feed into `display_size` with my current settings. I thought it might be possible to build a `pretty_table` around a given number of rows rather than backing out the number of rows to show in the table from the display size. 

![image](https://user-images.githubusercontent.com/65452054/100926694-d6a62300-34b1-11eb-9e78-482462ec4e8d.png)

**Comment by ronisbr** [comment]
I see. Ok, makes sense. I just need to see how can I make this compatible with the entire API. We will have options that does the same thing.

**Comment by ronisbr** [comment]
The version 3.0.0 that I have just released can do that!
